Framework/Algorithms/src/ExtractFFTSpectrum.cpp

```diff
@@ -64,23 +82,54 @@ void ExtractFFTSpectrum::exec() {
                                                   inputImagWS);
       childFFT->setProperty<int>("Imaginary", i);
     }
+    childFFT->setProperty<double>("Shift", shift);
+    childFFT->setProperty<bool>("AutoShift", autoShift);
+    childFFT->setProperty<bool>("AcceptXRoundingErrors", xRoundingErrs);
     childFFT->execute();
     MatrixWorkspace_const_sptr fftTemp =
         childFFT->getProperty("OutputWorkspace");
+    unit = fftTemp->getAxis(0)->unit();
 
-    outputWS->setHistogram(i, fftTemp->histogram(fftPart));
+    const auto &histo = fftTemp->histogram(fftPart);
+    if (!inputImagWS) {
```

Comment:
I think this should be if the `FFTPart` is 0 - 2 as well as no imaginary workspace as these will be the spectra with a tail of zeros because they only contain positive frequencies.